mllib/src/test/scala/org/apache/spark/ml/regression/DecisionTreeRegressorSuite.scala

```diff
@@ -96,6 +108,15 @@ class DecisionTreeRegressorSuite
       assert(variance === expectedVariance,
         s"Expected variance $expectedVariance but got $variance.")
     }
+
+    val toyDF = TreeTests.setMetadata(toyData, Map.empty[Int, Int], 0)
+    dt.setMaxDepth(1)
+      .setMaxBins(6)
+      .setSeed(0)
+    val expectVariances = dt.fit(toyDF).transform(toyDF).select("variance").collect().map {
+      case Row(variance: Double) => variance }
+    val trueVariances = Array(0.667, 0.667, 0.667, 2.667, 2.667, 2.667)
+    trueVariances.zip(expectVariances).foreach(x => x._1 ~== x._2 absTol 1e-3)
```

Comment:
Although this technically works, it is less confusing if use `assert` and unpack the tuple. Like

```scala
...foreach { case (actual, expected) =>
  assert(actual ~== expected absTol 1e-3)
}
```